### PR TITLE
Minor Streaming buffer cleanups/v1

### PR DESCRIPTION
**Pull request by inashivb**

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 17510

File: src/util-streaming-buffer.c [hunk 1]
```diff
@@ -334,7 +334,10 @@ static int WARN_UNUSED SBBInit(StreamingBuffer *sb, const StreamingBufferConfig
     }
     sbb->offset = sb->region.stream_offset;
     sbb->len = sb->region.buf_offset;
-    (void)SBB_RB_INSERT(&sb->sbb_tree, sbb);
+    if (SBB_RB_INSERT(&sb->sbb_tree, sbb) != NULL) {
```
[the remark this answers is not in the document]
Comment (victorjulien) [reply src/util-streaming-buffer.c:337]: This should only be called if the tree is empty. Can we fail in that case?

[the remark this answers is not in the document]
Comment (inashivb) [reply src/util-streaming-buffer.c:337]: Indeed. I misread it as per its other call sites. I can't think of a reason we'll fail then but that doesn't mean we can't. 😅  Do you recommend I remove this check?

[the remark this answers is not in the document]
Comment (victorjulien) [reply src/util-streaming-buffer.c:337]: Did you review the insertion macro? Pretty sure we can't fail in an empty tree. So in that case adding error code that will never trigger is actually harmful

[the remark this answers is not in the document]
Comment (inashivb) [reply src/util-streaming-buffer.c:337]: I see. Thank you. Will remove the checks.

**Review by victorjulien — CHANGES_REQUESTED** [review]
see older inline stuff

**Pull request closed without merging**